src/test/codegen/link-dead-code.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// compile-flags:-Clink-dead-code
+
+#![feature(const_fn)]
+#![crate_type = "rlib"]
+
+// This test makes sure that, when -Clink-dead-code is specified, we generate
+// code for functions that would otherwise be skipped.
+
+// CHECK-LABEL: define hidden i64 @_ZN14link_dead_code8const_fn
+const fn const_fn() -> usize { 1 }
```

Comment:
Please change the `usize` to some more portable type. This test is failing on 32-bit platforms.

```
[00:52:46] ---- [codegen] codegen/link-dead-code.rs stdout ----
[00:52:46] 	
[00:52:46] error: verification with 'FileCheck' failed
[00:52:46] status: exit code: 1
[00:52:46] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/llvm/build/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/link-dead-code.ll" "/checkout/src/test/codegen/link-dead-code.rs"
[00:52:46] stdout:
[00:52:46] ------------------------------------------
[00:52:46] 
[00:52:46] ------------------------------------------
[00:52:46] stderr:
[00:52:46] ------------------------------------------
[00:52:46] /checkout/src/test/codegen/link-dead-code.rs:19:17: error: expected string not found in input
[00:52:46] // CHECK-LABEL: define hidden i64 @_ZN14link_dead_code8const_fn
[00:52:46]                 ^
[00:52:46] /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/link-dead-code.ll:1:1: note: scanning from here
[00:52:46] ; ModuleID = 'link_dead_code0-8787f43e282added376259c1adb08b80.rs'
[00:52:46] ^
[00:52:46] /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/link-dead-code.ll:8:1: note: possible intended match here
[00:52:46] define hidden i32 @_ZN14link_dead_code8const_fn17h307759a84b38b8b4E() unnamed_addr #0 {
[00:52:46] ^
[00:52:46] 
[00:52:46] ------------------------------------------
[00:52:46] 
[00:52:46] thread '[codegen] codegen/link-dead-code.rs' panicked at 'explicit panic', /checkout/src/tools/compiletest/src/runtest.rs:2776:8
[00:52:46] note: Run with `RUST_BACKTRACE=1` for a backtrace.
```